src/docker/shared.rs

```diff
@@ -537,37 +516,39 @@ pub(crate) fn docker_mount(
             None => env::var(var),
         };
 
+        // NOTE: we use canonical paths on the host, since it's unambiguous.
+        // to avoid changing paths on the mount, we use absolute paths.
```

Comment:
this note does not make much snse to me, specifically `to avoid changing paths`, what does "changing paths" mean?

Reply:
It means the path canonicalization without the symlinks expected for them on the host can change the mounted path and cause code to break. It's meant to convey the same thing as in https://github.com/cross-rs/cross/issues/920#issue-1298118330. I'll make it a bit clearer.

Reply:
I've updated the comment with an example and clearer wording:

```rust
// NOTE: we use canonical paths on the host, since it's unambiguous.
// however, for the mounted paths, we use the same path as was
// provided. this avoids canonicalizing symlinks which then causes
// the mounted path to differ from the path expected on the host.
// for example, if `/tmp` is a symlink to `/private/tmp`, canonicalizing
// it would lead to us mounting `/tmp/process` to `/private/tmp/process`,
// which would cause any code relying on `/tmp/process` to break.
```